### ISSUE TITLE
Using Pretrained Model to Finetune Librispeech gives an error.

### ISSUE THREAD

**Issue by ahsanmemon** [pytorch/fairseq#2887]
Getting possibly a dataset-related error over Finetuning wav2vec2.0 

Here are the steps that I followed
1. created conda environment
2. `git clone https://github.com/pytorch/fairseq`
3. `cd fairseq`
4. `pip install --editable ./`
5. `pip install soundfile`
6. created tsv files using: `python examples/wav2vec/wav2vec_manifest.py /home/ahsanmemon/Desktop/librispeech/LibriSpeech/train-clean-100/ --dest ~/Documents/fairseq/ahsan --ext flac --valid-percent 0.01`
7. downloaded model `wav2vec_small_960h.pt` and `dict.ltr.txt`
8. generate train labels using `split=train
python examples/wav2vec/libri_labels.py /home/ahsanmemon/Documents/fairseq/ahsan/train.tsv --output-dir /home/ahsanmemon/Documents/fairseq/ahsan --output-name $split`
8. generate valid labels using `split=valid
python examples/wav2vec/libri_labels.py /home/ahsanmemon/Documents/fairseq/ahsan/train.tsv --output-dir /home/ahsanmemon/Documents/fairseq/ahsan --output-name $split`
9. FineTune model using the following code
`valid_subset=valid`
`python train.py /home/ahsanmemon/Documents/fairseq/ahsan --save-dir /home/ahsanmemon/Documents/fairseq/ahsan/finetuned_model \
--post-process letter --valid-subset $valid_subset --no-epoch-checkpoints --best-checkpoint-metric wer --num-workers 1 \
--max-update 80000 --sentence-avg --task audio_pretraining --arch wav2vec_ctc --w2v-path /home/ahsanmemon/Documents/fairseq/ahsan/model/wav2vec_small_960h.pt \
--labels ltr --apply-mask --mask-selection static --mask-other 0 --mask-length 10 --mask-prob 0.5 --layerdrop 0.1 \
--mask-channel-selection static --mask-channel-other 0 --mask-channel-length 64 --mask-channel-prob 0.5 --zero-infinity \
--feature-grad-mult 0.0 --freeze-finetune-updates 10000 --validate-after-updates 10000 --optimizer adam \
--adam-betas '(0.9, 0.98)' --adam-eps 1e-08 --lr 2e-05 --lr-scheduler tri_stage --warmup-steps 8000 --hold-steps 32000 \
--decay-steps 40000 --final-lr-scale 0.05 --final-dropout 0.0 --dropout 0.0 --activation-dropout 0.1 --criterion ctc \
--attention-dropout 0.0 --max-tokens 1280000 --seed 2337 --log-format json --log-interval 500 --ddp-backend no_c10d
`

But I am getting this error:
![image](https://user-images.githubusercontent.com/28605862/98941815-e4016a80-250e-11eb-8189-5d6ea32caab8.png)

#### What have I tried?
 - checked tsv files and they seem to be fine
 - checked wrd and ltr files and they seem to be fine too (not sure how they are supposed to be but they have data in them)
![image](https://user-images.githubusercontent.com/28605862/98942402-bf59c280-250f-11eb-8372-395f55973add.png)

#### What's your environment?
 - torch version (1.7.0)

**Comment by alexeib** [comment]
can you share the full log? also can you print what w2v_model is? it should be of type Wav2Vec2Model which takes the following args in forward method:
`def forward(self, source, padding_mask=None, mask=True, features_only=False):`

but it looks like you are calling some other model that takes fewer arguments in the forward method? (or with different names)

**Comment by ahsanmemon** [comment]
Thank you for the prompt response, here is the log

I believe, by log you meant the output that gets printed on the console.

[20201113 - log](https://github.com/pytorch/fairseq/files/5535093/20201113.-.log)

**Comment by alexeib** [comment]
looks like you are trying to finetune a model that has already been finetuned:

```(w2v_encoder): Wav2VecEncoder(
    (w2v_model): Wav2VecCtc(
      (w2v_encoder): Wav2VecEncoder(
        (w2v_model): Wav2Vec2Model(
```

you are meant to provide a path to a non-finetuned model as w2v_path
if you want to continue finetuning an existing model use --restore-file (this keeps optimizer states and so on) or --finetune-from-model which resets optimizer states

 in this case you should use
